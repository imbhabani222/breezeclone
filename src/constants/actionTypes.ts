import keyMirror from "keymirror";

const actionTypes = {
  FETCH_TODO_REQUEST: null,
  FETCH_TODO_SUCCESS: null,
  FETCH_TODO_FAILURE: null,
  SAMPLE_ACTION: null,
  ACTIONS: null,
  FETCH_IDENTITY: null,
  FETCH_IDENTITY_SUCCESS: null,
  FETCH_IDENTITY_FAILURE: null,
  FETCH_TEST_DETAILS: null,
  FETCH_TEST_DETAILS_SUCCESS: null,
  FETCH_TEST_DETAILS_FAILURE: null,
  FETCH_SERVER_TIME_REQUEST: null,
  FETCH_SERVER_TIME_SUCCESS: null,
  FETCH_SERVER_TIME_FAILURE: null,

  PROBLEM_DETAILS_REQUEST: null,
  PROBLEM_DETAILS_SUCCESS: null,
  PROBLEM_DETAILS_FAILURE: null,

  FETCH_QUESTIONS_REQUEST: null,
  FETCH_QUESTIONS_SUCCESS: null,
  FETCH_QUESTIONS_FAILURE: null,

  DELETE_ANSWER_REQUEST: null,
  DELETE_ANSWER_SUCCESS: null,
  DELETE_ANSWER_FAILURE: null,

  DELETE_FIB_REQUEST: null,
  DELETE_FIB_SUCCESS: null,
  DELETE_FIB_FAILURE: null,

  FETCH_PROBLEM_REQUEST: null,
  FETCH_PROBLEM_SUCCESS: null,
  FETCH_PROBLEM_FAILURE: null,

  POST_TEST_REQUEST: null,
  POST_TEST_SUCCESS: null,
  POST_TEST_FAILURE: null,

  FETCH_SINGLE_PRB_SOL_REQUEST: null,
  FETCH_SINGLE_PRB_SOL_SUCCESS: null,
  FETCH_SINGLE_MCQ_SOL_FAILURE: null,

  POST_SINGLE_PRB_SOL_REQUEST: null,
  POST_SINGLE_PRB_SOL_SUCCESS: null,
  POST_SINGLE_PRB_SOL_FAILURE: null,

  POST_FEEDBACk_REQUEST: null,
  POST_FEEDBACK_SUCCESS: null,
  POST_FEEDBACK_FAILURE: null,

  FETCH_SOLUTIONSET_REQUEST: null,
  FETCH_SOLUTIONSET_SUCCESS: null,
  FETCH_SOLUTIONSET_FAILURE: null,

  PATCH_FIB_REQUEST: null,
  PATCH_FIB_REQUEST_SUCCESS: null,
  PATCH_FIB_REQUEST_FAILURE: null,

  POST_REPORT_ISSUE_REQUEST: null,
  POST_REPORT_ISSUE_SUCCESS: null,
  POST_REPORT_ISSUE_FAILURE: null,

  PATCH_ANSWER_REQUEST: null,
  PATCH_ANSWER_REQUEST_SUCCESS: null,
  PATCH_ANSWER_REQUEST_FAILURE: null,

  POST_UPLOAD_FILE_REQUEST: null,
  POST_UPLOAD_FILE_SUCCESS: null,
  POST_UPLOAD_FILE_FAILURE: null,

  POST_SECTION_REQUEST: null,
  POST_SECTION_SUCCESS: null,
  POST_SECTION_FAILURE: null,

  GET_LANGUAGE_REQUEST: null,
  GET_LANGUAGE_SUCCESS: null,
  GET_LANGUAGE_FAILURE: null,

  RUN_CODE_CUSTOM_INPUT_REQUEST: null,
  RUN_CODE_CUSTOM_INPUT_SUCCESS: null,
  RUN_CODE_CUSTOM_INPUT_FAILURE: null,

  RUN_CODE_CUSTOM_INPUT_RESULT: null,

  TESTCASES_RUN_CODE__REQUEST: null,
  TESTCASES_RUN_CODE__SUCCESS: null,
  TESTCASES_RUN_CODE__FAILURE: null,
};

export default keyMirror(actionTypes);
