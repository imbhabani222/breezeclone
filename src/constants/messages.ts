const messages = {
  NOT_AUTHORIZED:
    "You are not authorized or session has been expired. Please try again.",
  NO_DATA_AVAILABLE: "No data available.",
  ACCESS_DENIED: "Access Denied/Forbidden.",
  DEFAULT_ERROR_MESSAGE:
    "There seems to be a problem, please retry after sometime.",
  USER_NOT_FOUND: "Invalid user",
  IMAGE_NOT_AVAILABLE: "Image not available",
  ARE_YOU_SURE: "Are you sure?",
  TYPE_OUTLINE: "outline",
  TYPE_RECTFILLED: "rectFilled",
  TYPE_FILLED: "filled",
  PLEASE_TYPE_ANSWER: "Please type your answer.",
  EXAM_NAME: "[0-2 years] Data Analytics",
  EXAM_TYPE: "By DoSelect UI/UX Testing",
  MODAL_CONTENT:
    "# Are you sure ? \n You still have **%hours% hours %minutes% mins %seconds% seconds** left to submit this assessment. </br></br>You have attempted %completed% of %total% questions.</br></br>You will not be able to attempt any question or make changes to your solution once you submit.",
  LOADING: "Loading...",
  PROCEED_TO_FEEDBACK: "PROCEED TO FEEDBACK PAGE",
  GOING_GREAT:
    "Going great .</br></br>You have attempted %completed% of %total% questions.",
  AUTO_NEXT_MODAL:
    "This section has been auto-submitted as the allotted time is up.</br></br>You have attempted %completed% of %total% questions.",
  BROWSER_TOLERANCE_INFO:
    "  You had navigated away from the test screen. This incident will be reported to the test administrator.",
  LOAD_MORE: "Load More",
  BACKBUTTON: " Go back to Assessment",
  PLATFORM_SUPPORT: "Platform Support",
  CLIENTS_SUPPORT: "Clients Support",
  HELP: "Help",
  QUESTION: "View All Questions",
  RIGHT: "right",
  FEEDBACK: {
    FEEDBACKCOMPLETED: "Woot! You've completed this assessment.",
    FEEDBACKTITLE: "SHARE FEEDBACK",
    FEEDBACKNOTIFIED:
      "We've notified team %orgName% about this. If you have any queries regarding this, please get in touch with your point of contact.",
    SENDFEEDBACK: "Send Feedback",
    ENTERFEWWORDS: "Please enter few words",
    MESSAGE_LABEL: "message",
    SUBMIT_LABEL: "Submit",
    FEEDBACKFORM: "feedbackForm",
    LIKETOTELL: "Is there something that you'd like to tell us?",
  },
  REPORT_LABLE: "REPORT AN ISSUE",
  REPORT_MODAL_TITLE: "Report an issue with Stack Using Interface",
  REPORT_MODAL_INPUT_LABEL: "  Other:",
  PROGRESS_DATABAR_TEXT: "%completed%/%total% Questions attempted",
  RUN_CODE_LABEL: "Run Code",
  ABORT_LABEL: "Abort",
  TEST_RESULTS_LABEL: "Test results",
  TEST_RESULTS_TEXT: 'Click on "Run code" to evaluate your solution',
  CUSTOM_INPUT_LABEL: "Custom Input",
  CUSTOM_INPUT_TEXT: "STDIN",
  ICON: "Icon",
  TIME_LABEL: {
    H: "H:",
    M: "M:",
    S: "S",
  },
  FAQ: {
    KEY1: "1",
    KEY2: "2",
    KEY3: "3",
    TAB1: "Help",
    TAB2: "Coding Environment",
    TAB3: "Status Code",
    SEARCH_PLACEHOLDER: "Ask a question",
    FAQINFOTEXT: `Please find below the general instructions and information about 
    DoSelect's assessment environment. We highly recommend that you go
    through this before you start taking the assessment.`,
  },
  TESTSTATUS: {
    ICONSTATUS: "iconsStatus",
    ACTIVATEICON: "activateIcon",
    DEACTIVATEICON: "deactivateIcon",
  },
  DESCRIPTION_LABEL: "Description",
  STATUS_CODE_LABEL: "Status Code",
  LIBRARIES_AVAILABLE_LABEL: "Libraries Available",
  TIME_LIMIT_LABEL: "Time Limit(in seconds)",
  MEMORY_UNIT_LABEL: "Memory Limit (in MB)",

  LIBRARIES_AVAILABLE_KEY: "Libraries available",
  TIME_LIMIT_KEY: "Time limit (in seconds)",
  MEMORY_UNIT_KEY: "Memory limit (in MB)",

  VERSION_LABEL: "Version",
  LANGUAGE_LABEL: "Language",
  DO_SELECT_RESEVE_RIGHTS: "Doselect. All rights reserved",
  LOGO_LABEL: "Logo",
  TEXT_LABEL: "text",
  TIME_FOR_EXAM_LABEL: "%hours% hours %minutes% mins",
  TIME_FOR_MODAL: "%hours% hours %minutes% mins %seconds% seconds",
  ALT_IMG: "icon Doselect",
  CANCEL: "Cancel",
  SUBMIT_END: "Submit and end the Assessment",
  SUBMIT_MOVE_NEXT: "Submit and Move to Next Section",
  PROCEED_TO_NEXT: "Proceed to next section",
  UPLOAD: "Upload",
  SAVE: "Save",
  QUESTION_TYPE: "MCQ + Coding + Fill in the  Blanks",
  LOGO: "do select logo",
  CAMERA: "Video Proctoring",
  NETWORK: "Network quality",
  TIMER: "Timer",

  BROWSER_FULLSCREEN: `<p>This %type% requires your browser window to be in fullscreen mode</p>`,
  UPLOAD_ATTACHMENT: "Upload an attachment with this solution.",
  VIDEO_TEST: {
    DESC: "Description",
    VID_DUR: "Video Duration",
    SEC: "seconds",
    RECORD: "Record",
    STOP: "Stop",
    RESUME: "Resume",
    PAUSE: "Pause",
    SUBMIT: "Submit",
    REC_DUR: "Record duration:",
  },

  CODING: {
    CODING: "Coding",
    RELOAD_STUB: "Reload Code Stub?",
    RELOAD_STUB_CONTENT: `You have changed the technology of this solution. Do you want
    to REPLACE your current code with the default code stub of
    this technology? Please note that you will lose your current
    code by doing this.`,
    NO_TEST_FOUND: "No Test Cases Found",
    RUN_ALL_TEST_CASES: "Run all Test Cases",
    DESCRIPTION: "Description",
    DESCRIPTIONS: "Descriptions",
    INPUT: "Input",
    OUTPUT: "Output",
    EXPECTED_OUTPUT: "Expected Output",
    EVAL_TIME: "Eval Time",
    CPU: "CPU",
    MEMORY: "Memory",
    EXECUTION_LOGS: "Execution Logs",
    ANNOTATION: "Annotation",
  },
};

export default messages;
