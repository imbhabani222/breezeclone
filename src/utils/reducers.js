/* eslint-disable no-prototype-builtins */
export const createReducer = (initialState, handlers) => {
  return function reducer(state = initialState, action) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action);
    } else {
      return state;
    }
  };
};

export const getFetchingState = (state) => ({
  ...state,
  fetching: true,
  result: null,
  error: null,
});

export const getSuccessState = (state, { result }) => ({
  ...state,
  fetching: false,
  result: result.result || result,
  error: null,
});

export const getErrorState = (state, { error }) => ({
  ...state,
  fetching: false,
  result: null,
  error,
});

export const getResetState = (state) => ({
  ...state,
  fetching: false,
  result: null,
  error: null,
});

export const getActionInitialState = (state) => ({
  ...state,
});

export const getActionDefaultState = (state, { result: { data } }) => ({
  ...state,
  ...data,
});

export const checkAction = (state = {}, action) => {
  switch (action.type) {
    case "CHECK_ACTION": {
      console.log(state, action);
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const customResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_CUSTOM_RESULT": {
      console.log(state, action, "custommmmmmmmmm");
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};

export const testCasesResult = (state = {}, action) => {
  switch (action.type) {
    case "GET_TESTCASES_RESULT": {
      return {
        data: { ...state.data, ...action.data },
      };
    }
    default:
      return state;
  }
};
