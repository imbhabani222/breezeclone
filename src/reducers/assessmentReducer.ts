import actionTypes from "../constants/actionTypes/assessmentActionsTypes";

import {
  assessmentDetails,
  assessmentState,
  IdentityActions,
  TestDetailsActions,
  IdentityState,
  TechnologyActions,
} from "../types/types";

const initialIdentity: IdentityState = {
  pending: false,
  identity: [],
  error: null,
  timeLeft: Date.now() + 904444,
  declarationStatus: false,
  modalStatus: false,
};
const initialState: assessmentState = {
  loading: false,
  details: {},
  error: null,
};
export const getIdentity = (
  state = initialIdentity,
  action: IdentityActions
) => {
  switch (action.type) {
    case actionTypes.FETCH_IDENTITY:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.FETCH_IDENTITY_SUCCESS:
      return {
        ...state,
        loading: false,
        identity: action.payload.identity,
        error: null,
      };
    case actionTypes.FETCH_IDENTITY_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getTestDetails = (
  state = initialState,
  action: TestDetailsActions
) => {
  switch (action.type) {
    case actionTypes.TEST_DETAILS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.TEST_DETAILS_SUCCESS:
      return {
        ...state,
        loading: false,
        details: action.payload.details,
      };
    case actionTypes.TEST_DETAILS_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};

export const getTechnologyDetails = (
  state = initialState,
  action: TechnologyActions
) => {
  switch (action.type) {
    case actionTypes.TECHNOLOGY_DETAILS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case actionTypes.TECHNOLOGY_DETAILS_SUCCESS:
      return {
        ...state,
        loading: false,
        details: action.payload.details,
      };
    case actionTypes.TECHNOLOGY_DETAILS_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return {
        ...state,
      };
  }
};
