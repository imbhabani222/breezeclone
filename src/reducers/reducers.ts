import actionTypes from "../constants/actionTypes";

import { TodoActions, TodoState } from "../types/types";

const initialState: TodoState = {
  pending: false,
  todos: [],
  error: null
};

// eslint-disable-next-line import/no-anonymous-default-export
export default (state = initialState, action: TodoActions) => {
  switch (action.type) {
    case actionTypes.FETCH_TODO_REQUEST:
      return {
        ...state,
        pending: true
      };
    case actionTypes.FETCH_TODO_SUCCESS:
      return {
        ...state,
        pending: false,
        todos: action.payload.todos,
        error: null
      };
    case actionTypes.FETCH_TODO_FAILURE:
      return {
        ...state,
        pending: false,
        todos: [],
        error: action.payload.error
      };
    default:
      return {
        ...state
      };
  }
};
