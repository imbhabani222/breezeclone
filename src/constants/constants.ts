import Cookies from "universal-cookie";
import messages from "./messages";

import { useLocation } from "react-router-dom";
const cookies = new Cookies();
const csrfToken = cookies.get("doselectcsrf");

const constants = {
  HTTP_REQ_DEFAULT_HEADERS: {
    Accept: "application/json",
    "Content-Type": "application/json; charset=UTF-8",
  },
  HTTP_POST_HEADERS: {},
  HTTP_MULTIPART_FORM_HEADERS: {
    "Content-Type": false,
  },
  HTTP_METHODS: {
    POST: "POST",
    GET: "GET",
    DELETE: "DELETE",
    PATCH: "PATCH",
    PUT: "PUT",
  },
  HTTP_REQ_ERROR_MAP: {
    401: { status: 401, message: messages.NOT_AUTHORIZED },
    403: { status: 403, message: messages.ACCESS_DENIED },
    404: { status: 404, message: messages.NO_DATA_AVAILABLE },
  },
  ROUTE: {
    LANDING_PAGE: "/test/:slug/landingPage",
    INSTRUCTION_PAGE: "/test/:slug/instructionPage",
    DOSELECT_INSTRUCTION: "/test/:slug/doSelectInstruction",
    TEST_SECTION_SUBMIT: "/test/:slug/problems",
    FAQ: "/test/:slug/Faqs",
    FEEDBACK: "/test/:slug/Feedback",
    TESTCOMPLETE: "/test/:slug/TestComplete",
    TESTLAYOUT: "/test/:slug/TestLayout",
    RICH_TEXT_EDITOR: "/test/:slug/problems/id",
    FIll_IN_BLANKS: "/tegst/:slug/problems/:id",
    MCQ_PAGE: "/tegst/:slug/problems/:id",
    SWITCH_PATH: "/test/:slug/problems/:id",
  },
  NAME_KEY: "name",
  STATUS_KEY: "status",
  LANGUAGE_KEY: "Language",
  VERSION_KEY: "Version",
  MU_KEY: "Mu",
  TL_KEY: "Tl",
  LA_KEY: "LA",
  DESCRIPTION_KEY: "description",
  DEFAULT_KEY: "default",
  VIDEO_KEY: "video",
  PROGRESSBAR_KEY: "progressbar",
};

export default constants;
