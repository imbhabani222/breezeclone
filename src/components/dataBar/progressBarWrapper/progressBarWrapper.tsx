import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";

import PropTypes from "prop-types";
import { Progress, Typography } from "antd";
import styles from "../dataBar.module.scss";
import colorVars from "../../../commonStyles/color.module.scss";
import messages from "../../../constants/messages";
import { stringFiller } from "../../../data/utils";
import { getProblemSelector } from "../../../selectors/problemSelector";

const { Text } = Typography;
const { PROGRESS_DATABAR_TEXT } = messages;

// const getAllData = () => {
//   return stringFiller(PROGRESS_DATABAR_TEXT, { completed, total });
// };

const ProgressBarWrapper = () => {
  const problems = useSelector(getProblemSelector);

  console.log(problems, "Problems");
  let total: number = 0,
    completed: any = 0;

  if (problems?.length >= 0) {
    problems &&
      problems.forEach((sec: any) => {
        sec.problems.forEach((singleObj: any) => {
          total++;
          if (singleObj.solution !== null) {
            completed++;
          }
        });
      });
  }

  const getText = () => {
    return stringFiller(PROGRESS_DATABAR_TEXT, { completed, total });
  };

  const [percent, setPercent] = useState<number>((completed / total) * 100);

  useEffect(() => {
    setPercent((completed / total) * 100);
  }, [completed, total]);

  return (
    <div className={styles.progressbar_col}>
      <Progress
        percent={percent}
        strokeWidth={+styles.progress_stroke_width || 10}
        strokeColor={colorVars.progress_stroke_color}
        trailColor={colorVars.progress_unfilled_color}
        showInfo={false}
        className={styles.progress}
      />
      <Text className={styles.progressbar_text} title="databartitle">
        {getText()}
      </Text>
    </div>
  );
};

export default React.memo(ProgressBarWrapper);
