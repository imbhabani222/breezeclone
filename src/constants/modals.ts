import { useState } from "react";
import messages from "./messages";

import useFetch from "./../data/dataBar-async";

import { getDuration, getTimeDiff, stringFiller } from "../data/utils";
import ProgressBarWrapper from "./../components/dataBar/progressBarWrapper/progressBarWrapper";
import { getProblemSelector } from "../selectors/problemSelector";
import { useSelector } from "react-redux";

const {
  FEEDBACK: { SUBMIT_LABEL },
  CANCEL,
  SUBMIT_MOVE_NEXT,
  SUBMIT_END,
  ARE_YOU_SURE,
  REPORT_LABLE,
  PROCEED_TO_NEXT,
  MODAL_CONTENT,
  TYPE_OUTLINE,
  TYPE_FILLED,
  AUTO_NEXT_MODAL,
  PROCEED_TO_FEEDBACK,
  GOING_GREAT,
} = messages;

const GetString = (timeLeft: any, str: string) => {
  const problems = useSelector(getProblemSelector);
  let total: number = 0,
    completed: number = 0;

  if (problems?.length >= 0) {
    problems &&
      problems.forEach((sec: any) => {
        sec.problems.forEach((singleObj: any) => {
          total++;
          if (singleObj.solution !== null) {
            completed++;
          }
        });
      });
  }

  return stringFiller(str, {
    ...timeLeft,
    total: total,
    completed: completed,
  });
};

const GetNextString = (timeLeft: any, str: string) => {
  const problems = useSelector(getProblemSelector);
  let SectionTotal: number = 0,
    sectionCompleted: number = 0;

  if (problems.length >= 0) {
    let current =
      problems &&
      problems?.find((val: any) => {
        return val.status === "CTK";
      });
    current &&
      current?.problems.map((val: any) => {
        if (val.solution !== null) {
          sectionCompleted++;
        }
      });
    SectionTotal = current && current.problems.length;
  }
  // console.log({
  //   ...timeLeft,
  //   total: SectionTotal,
  //   completed: sectionCompleted,
  // });

  return (
    stringFiller(str, {
      ...timeLeft,
      total: SectionTotal,
      completed: sectionCompleted,
    }) || []
  );
};

export const getSubmitConformationModal = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: ARE_YOU_SURE,
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? GetString(timeLeft, MODAL_CONTENT) : text,
    buttons: contentType
      ? [
          {
            type: TYPE_OUTLINE,
            display: CANCEL,
            onClick: onCancel,
          },
          {
            type: TYPE_FILLED,
            display: SUBMIT_END,
            onClick: onOk,
          },
        ]
      : null,
  };
};

export const getSubmitNextConformationModal = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: ARE_YOU_SURE,
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? GetNextString(timeLeft, MODAL_CONTENT) : text,
    buttons: contentType
      ? [
          {
            type: TYPE_OUTLINE,
            display: CANCEL,
            onClick: onCancel,
          },
          {
            type: TYPE_FILLED,
            display: SUBMIT_MOVE_NEXT,
            onClick: onOk,
          },
        ]
      : null,
  };
};
export const autoRedirectModalToNext = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: "",
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? text : GetString(timeLeft, AUTO_NEXT_MODAL),
    buttons: contentType
      ? [
          {
            type: TYPE_FILLED,
            display: PROCEED_TO_FEEDBACK,
            onClick: onOk,
          },
        ]
      : null,
  };
};
export const autoRedirectToSection = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: "",
    onOk: onOk,
    onCancel: onCancel,
    text: text,
    buttons: contentType
      ? [
          {
            type: TYPE_FILLED,
            display: "Submit and move to next section",
            onClick: onOk,
          },
        ]
      : null,
  };
};
