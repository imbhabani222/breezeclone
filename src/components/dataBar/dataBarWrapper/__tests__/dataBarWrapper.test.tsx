import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import DataBarWrapper, { databarProps } from "../dataBarWrapper";
import progressBarWrapper from "../../progressBarWrapper/progressBarWrapper";
import constants from "../../../../constants/constants";
import ProgressBarWrapper from "../../progressBarWrapper/progressBarWrapper";
import { combineReducers, createStore } from "redux";
import { getTestDetails } from "../../../../reducers/assessmentReducer";
import { Provider } from "react-redux";
const { PROGRESSBAR_KEY } = constants;

describe("DatabarWrapper Testing", () => {
  const initialState = {
    pending: false,
    identity: [],
    error: null,
    timeLeft: 400,
  };

  const mock = {
    title: "apiTest",
    subTitle: "Hutech Dev",
    values: [
      {
        title: "apiTest",
        subTitle: "Hutech Dev",
      },
    ],
    type: "",
    image: "/logo.svg",
    alt: "some random string",
    className: "hutechlogo",
  };

  const fakeStore = createStore(
    combineReducers({ testDetails: getTestDetails }),
    {
      testDetails: initialState,
    }
  );
  fakeStore.dispatch = jest.fn((): any => {
    return mock;
  });

  const defaultProps: databarProps = {
    title: "",
    subTitle: "",
    image: "",
    alt: "",
    values: [],
    type: "",
    className: "",
  };

  function renderDatabarWrapper(props: Partial<databarProps> = {}) {
    return render(
      <Provider store={fakeStore}>
        <BrowserRouter>
          <DataBarWrapper {...defaultProps} {...props} />
        </BrowserRouter>
      </Provider>
    );
  }
  <DataBarWrapper {...{ ...defaultProps, title: "" }} />;
  afterEach(cleanup);
  test("Render without Crashing", () => {
    renderDatabarWrapper();
  });
  test("Match snapshot databarWrapper component", () => {
    const { asFragment } = renderDatabarWrapper();
    expect(asFragment()).toMatchSnapshot();
  });
  test("should not display databar wrapper", async () => {
    renderDatabarWrapper();
    expect(screen.queryByTitle(PROGRESSBAR_KEY)).not.toBeInTheDocument();
  });
  // test("should display databar wrapper", async () => {
  //   renderDatabarWrapper();
  //   expect(screen.queryByTitle(PROGRESSBAR_KEY)).not.toBeInTheDocument();
  // });
  test("display databar", async () => {
    renderDatabarWrapper(mock);
    expect(screen.getByTitle(mock.title)).toBeDefined();
    expect(screen.getByText(mock.subTitle)).toBeDefined();
  });
  test("renders the image src and alt name", () => {
    renderDatabarWrapper(mock);
    const image = screen.getByAltText("some random string");
    expect(screen.getByAltText(mock.alt)).toBeInTheDocument();
    expect(image).toHaveAttribute("src", "/logo.svg");
  });

  //Doubt
  test("Type image is string", () => {
    renderDatabarWrapper();
    // renderDatabarWrapper({ ...{ ...defaultProps, type: "" } });
    const logo = screen.getByTestId("imagediv");
    expect(logo).toHaveAttribute("src", "/logo.svg");
    expect(logo).toHaveAttribute("alt", "Logo");
  });
  //Doubt
  test("Type check for progressbar", () => {
    renderDatabarWrapper({ ...{ ...defaultProps, type: PROGRESSBAR_KEY } });
    const commentList = expect(screen.queryByTestId("databardiv")).toBe(
      <ProgressBarWrapper />
    );
    console.log(commentList, "common");
    expect(commentList).toBeTruthy();
  });
});

//EXAMPLES
// describe("Logo", () => {
//   test('Logo must have src = "/logo.svg" and alt = "Logo"', () => {
//     render(<Logo />);
//     const logo = screen.getByRole("img");
//     expect(logo).toHaveAttribute("src", "/logo.svg");
//     expect(logo).toHaveAttribute("alt", "Logo");
//   });
// });

// describe("<Image /> component", () => {
//   afterEach(cleanup);

// describe("Component as a whole", () => {
//   it("renders the image with a src, alt and a className ", () => {
//     const testProps = {
//       imageAlt: "some random string",
//       imageSrc: "" /* [ASK]: How to test this */,
//       className: "imageDescripton" /* [ASK]: How to test this */
//     };

//     const { getByAltText } = render(<Image {...testProps} />);
//     const { getByText } = render(<Image {...testProps} />);

//     const imageAltNode = getByAltText(testProps.imageAlt);
//     const imageClassNameNode = getByText(`${testProps.className}`); // [FAIL]: Fails with error.  Unable to find an element with the text: imageDescripton. Regex problem?

//     expect(imageAltNode).toBeDefined();
//     expect(imageClassNameNode).toBeDefined();
//   });
// });
// });
