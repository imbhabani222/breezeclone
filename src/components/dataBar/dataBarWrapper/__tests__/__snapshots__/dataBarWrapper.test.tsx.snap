// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabarWrapper Testing Match snapshot databarWrapper component 1`] = `
<DocumentFragment>
  <div
    class="databar_wrapper_container"
  >
    <div
      class="ant-image"
      data-testid="imagediv"
    >
      <img
        alt=""
        class="ant-image-img responsive_img"
        src=""
      />
    </div>
    <div
      class="calendar_double_line"
    />
  </div>
</DocumentFragment>
`;
